File: server/src/main/java/io/github/lumklar/sortrss/server/infrastructure/persistence/repository/UserJpaRepository.java
package io.github.lumklar.sortrss.server.infrastructure.persistence.repository;

import io.github.lumklar.sortrss.server.infrastruncture.persistence.entity.UserPO;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserJpaRepository extends JpaRepository<UserPO, Long> {
    UserPO findByUsername(String username);
    Boolean existsByUsername(String username);
}

File: server/src/main/kotlin/io/github/lumklar/sortrss/server/infrastructure/persistence/repository/jpa/UserJpaRepository.kt
package io.github.lumklar.sortrss.server.infrastructure.persistence.repository.jpa

import io.github.lumklar.sortrss.server.infrastructure.persistence.entity.UserPO
import org.springframework.data.jpa.repository.JpaRepository

interface UserJpaRepository : JpaRepository<UserPO, Long> {
    fun findByUsername(username: String?): UserPO?
    fun existsByUsername(username: String?): Boolean
}